sequence_processing.py

```diff
@@ -0,0 +1,143 @@
+    """
+Performs the following procedures with nucleotide sequences:
+    transcribe — returns the transcribed sequence
+    reverse — returns the reversed sequence
+    complement — returns the complementary sequence
+    reverse_complement — returns the reverse complementary sequence
+    :param args: str or list of sequence of nucleotides and name of function
+    :return:str or list of processed sequence
+    """
+    import additional_modules.run_dna_rna_tools
+    *sequence, function = args
+    if type(sequence) == list:
+        for nucleotides in sequence:
+            for nucleotide in nucleotides:
+                if nucleotide in additional_modules.run_dna_rna_tools.dna_rna_complement_dict.keys() and nucleotides in additional_modules.run_dna_rna_tools.dna_rna_complement_dict.values():
+                    raise ValueError('use correct sequence')
+                if ('U' or 'u') in nucleotides and (function == 'transcribe' or function == 'complement_transcribe'):
+                    raise ValueError('use correct sequence')
+    else:
+        for nucleotides in sequence:
+            if nucleotides in additional_modules.run_dna_rna_tools.dna_rna_complement_dict.keys() and nucleotides in additional_modules.run_dna_rna_tools.dna_rna_complement_dict.values():
+                raise ValueError('use correct sequence')
+            if ('U' or 'u') in nucleotides and (function == 'transcribe' or function == 'complement_transcribe'):
+                raise ValueError('use correct sequence')
+    if function == 'reverse':
+        return additional_modules.run_dna_rna_tools.reverse(sequence)
+    elif function == 'transcribe':
+        return additional_modules.run_dna_rna_tools.transcribe(sequence)
+    elif function == 'complement':
+        return additional_modules.run_dna_rna_tools.complement(sequence)
+    elif function == 'reverse_complement':
+        return additional_modules.run_dna_rna_tools.reverse_complement(sequence)
+
+
+def run_fastq(input_path: str, gc_bounds: Union[int, float, tuple] = (0, 100),
+              length_bounds: Union[int, float, tuple] = (0, 2**32), quality_threshold: int = 0, output_filename: str = 'input_filename.fastq') -> dict:
+    """
+This function filters reads based on the length of their nucleotide sequence, GC-content and phred33-score
+    :param input_path: path to your file. If you work in windows, add 'r' before 'path'
+    :param gc_bounds: GC-content interval (percentage) for filtering. Default = 0, 100
+    :param length_bounds: length interval for filtering. Default = 0, 2**32
+    :param quality_threshold: threshold value of average read quality for filtering. Default = 0
+    :param output_filename: name of file to output. Default: input_filename.fastq
+    :return: dictionary consisting of filtered fastq sequences matching all filters
+    """
+    import additional_modules.run_fastq
+    if output_filename == 'input_filename.fastq':
+        output_filename = input_path.split('\\')[-1]

```

Comment:
В целом ок, но видишь, у тебя это по сути будет работать только на винде:)
Чтобы было адаптивно:
```suggestion
        output_filename = os.path.basename(input_path)
```